### DISCUSSION TITLE
exporting to gp3, 4 or 5

### DISCUSSION THREAD

**Discussion by Dontwanno** [CoderLine/alphaTab#864] [Q&A] [answered]
Is there currently a way to export a loaded score to gp3, gp4 or gp5?

**Comment by Danielku15** [comment, marked as answer]
No currently the only exporter alphaTab has is Guitar Pro 7. Exporting to GP7 (and potentially GP6) is easier because internally the file is just an XML document and it is rather easy to reverse engineer and also write. 

With Guitar Pro 3-5 we have a binary format where not all bytes are known. But with some effort I guess it would be possible to create an exporter for older formats too.

**Reply by Dontwanno** [reply]
Ah, oke. Because I did find this website https://tabplayer.online/ where you can upload any type of gp file and download It as gp5, but I can't figure out how theyre doing it.

**Comment by circinusX1** [comment]
Does this work ?
```
                const gexporter = new alphaTab.exporter.Gp7Exporter();
                const data = gexporter.export(api.score, api.settings);  // will return a Uint8Array
                if (1){ // download
                    const a = document.createElement('a');
                    a.download = api.score.title.length > 0 ? api.score.title + '.gp' : 'Title-Song.gp';
                    a.href = URL.createObjectURL(new Blob([data]));
                    document.body.appendChild(a);
                    a.click();
                    document.body.removeChild(a);
                }else{ // play
                    api.load(data);
                }
```

**Reply by Dontwanno** [reply]
That only converts to gp7, not to gp3, 4 or 5

**Reply by Danielku15** [reply]
Again: alphaTab does **not** support gp3,4,5 export. We only offer features to export to GP7. This website you found must be doing it somehow different, but not using alphaTab.
